nixos/modules/services/networking/go-away.nix

```
@@ -0,0 +1,257 @@
+{
+  config,
+  lib,
+  pkgs,
+  utils,
+  ...
+}:
+let
+  inherit (lib) types;
+  yamlFormat = pkgs.formats.yaml { };
+
+  cfg = config.services.go-away;
+  hostConfig = config;
+  enabledInstances = lib.filterAttrs (_: conf: conf.enable) cfg.instances;
+
+  instanceName = name: if name == "" then "go-away" else "go-away-${name}";
+
+  configFile = name: instance: yamlFormat.generate "${instanceName name}-config.yaml" instance.config;
+  policyEtcPath = name: "go-away/${if name == "" then "default" else name}-policy.yaml";
+  policyFile =
+    name: instance:
+    pkgs.runCommandLocal "${instanceName name}-policy.yaml"
+      {
+        nativeBuildInputs = [ cfg.package ];
+      }
+      ''
+        cat ${yamlFormat.generate "${instanceName name}-policy.yaml" instance.policy} > $out
+        go-away -check -config ${configFile name instance} -policy $out
+      '';
+in
+{
+  options.services.go-away = {
+    package = lib.mkPackageOption pkgs "go-away" { };
+
+    instances = lib.mkOption {
+      default = { };
+      description = ''
+        An attribute set of Go-Away instances.
+
+        The attribute name may be an empty string, in which case the `-<name>` suffix is not added to the service name
+        and socket paths.
+      '';
+      type = types.attrsOf (
+        types.submodule (
+          { name, config, ... }:
+          {
+            options = {
+              enable = lib.mkEnableOption "this instance of Go-Away" // {
+                default = true;
+              };
+
+              extraFlags = lib.mkOption {
+                default = [ ];
+                description = "A list of extra flags to be passed to Go-Away.";
+                example = [ "-slog-level DEBUG" ];
+                type = types.listOf types.str;
+              };
+
+              bindHost = lib.mkOption {
+                default = "";
+                type = types.str;
+              };
+
+              bindPort = lib.mkOption {
+                type = types.ints.positive;
+              };
+
+              useACMEHost = lib.mkOption {
+                default = null;
+                type = types.nullOr types.str;
+              };
+
+              tlsCertificate = lib.mkOption {
+                default =
+                  if config.useACMEHost != null then
+                    "${hostConfig.security.acme.certs.${config.useACMEHost}.directory}/fullchain.pem"
+                  else
+                    null;
+                type = types.nullOr types.str;
+              };
+
+              tlsKey = lib.mkOption {
+                default =
+                  if config.useACMEHost != null then
+                    "${hostConfig.security.acme.certs.${config.useACMEHost}.directory}/key.pem"
+                  else
+                    null;
+                type = types.nullOr types.str;
+              };
+
+              config = lib.mkOption {
```

```suggestion
              settings = lib.mkOption {
```

We call this settings per RFC 42.